src/Host/Broker/Impl/Startup/CommonStartup.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Broker.Logging;
+using Microsoft.R.Host.Broker.Security;
+using Microsoft.R.Host.Protocol;
+using Newtonsoft.Json;
+using System.Diagnostics;
+using System.ServiceProcess;
+
+namespace Microsoft.R.Host.Broker.Startup {
+    internal class CommonStartup {
+        private static readonly ILoggerFactory _loggerFactory = new LoggerFactory();
+        private static ILogger _logger;
+        private static readonly StartupOptions _startupOptions = new StartupOptions();
+        private static readonly SecurityOptions _securityOptions = new SecurityOptions();
+        private static readonly CancellationTokenSource _cts = new CancellationTokenSource();
+
+        internal static IConfigurationRoot Configuration { get; set; }
+
+        internal static CancellationToken CancellationToken => _cts.Token;
+
+        internal static void CommonStartupInit(IConfigurationRoot configuration) {
```

Comment:
This was stuff in Program.Main, I moved it out of there so it could be accessed from service startup.